### ISSUE TITLE
Microsoft.NET.Sdk.IL change from PackageVersion => Version suffix broke WinForms usage

### ISSUE THREAD

**Issue by dagood** [dotnet/runtime#1338]
https://github.com/dotnet/winforms/pull/2533#issuecomment-570955502

> @dagood the PR is still failing to build due to an unresolved dependency
> 
> ```
> ##[error]src\Accessibility\src\Accessibility.ilproj(0,0): error NU1102: Unable to find package runtime.win-x64.microsoft.netcore.ilasm with version (>= 5.0.0)
>   - Found 1178 version(s) in dotnet-core [ Nearest version: 5.0.0-alpha.1.20054.2 ]
>   - Found 792 version(s) in dotnet-coreclr [ Nearest version: 5.0.0-alpha1.19564.1 ]
>   - Found 9 version(s) in nuget.org [ Nearest version: 2.0.8 ]
>   - Found 0 version(s) in arcade
>   - Found 0 version(s) in dotnet-eng
> ```

This is because WinForms uses `MicrosoftNETCoreILAsmPackageVersion`, but https://github.com/dotnet/coreclr/pull/27606 changed the IL SDK to use `MicrosoftNETCoreILAsmVersion`.

https://github.com/dotnet/runtime/blob/6425b2d1c2568b115a7a2b5cd130d85ec4f53a84/src/coreclr/src/.nuget/Microsoft.NET.Sdk.IL/targets/Microsoft.NET.Sdk.IL.targets#L30

I think it would be reasonable to accept both forms in the IL SDK. However, if it's better to just close this as "won't fix" and have WinForms change the property name instead, I think it will be useful to have had this issue open to track this breaking change.

**Comment by safern** [comment]
I think we should stick with the current property name as we've been transition Versions.props to not contain `PackageVersion` but `Version` as a suffix only. It is cleaner and easier to read, and also `PackageVersion` is redundant. The reason why `PackageVersion` was supported at the beginning was because DARC had a bug where it only matched the property if the suffix was `PackageVersion` instead of `Version`

**Comment by dagood** [comment]
> we've been transition Versions.props to not contain `PackageVersion` but `Version` as a suffix only.

Are all downstream users of `Microsoft.NET.Sdk.IL` also making this change? Is it ok to force this, and not have a clear error message?

**Comment by safern** [comment]
I don't know which are the users, but when we were still in corefx we made this change. This was done 2 months ago, so I would've expected other people to be broken by this already:

https://github.com/dotnet/runtime/commit/cdc33397f13075ab1e66baa2400cc4d8f8bb2a5b

I think we can error out if `PackageVersion` is set and `Version` is not or at least emit a warning.

**Comment by dagood** [comment]
Oh huh. This breaking change was absorbed in the past by https://github.com/dotnet/winforms/pull/2278, but the fix was reverted by https://github.com/dotnet/winforms/pull/2497 to go down to a version before this change happened. /cc @RussKie

Anyway, taking the breaking change again passed winforms CI: https://github.com/dotnet/winforms/pull/2647. Just waiting for a winforms reviewer.

Having a reasonable error message would have made this really easy to resolve and had WinForms deps flowing 10 days ago, FWIW. But now that it's over with, I don't know if it'll be hit again in the future.

**Comment by safern** [comment]
I guess we can close then? Or are you planning on putting up a PR to error/warn?

**Comment by dagood** [comment]
It's up to whoever makes decisions about `Microsoft.NET.Sdk.IL` to decide if the package should have these warnings/errors or not, IMO. I don't know enough about how/where this package is used to know for sure.

**Comment by RussKie** [comment]
It would good be make a  note of this for the future - add better error messages for breaking changes.

@dagood fixed us now (thank you!), so no further actions are required from our side.

**Comment by safern** [comment]
cc: @jkoritzinsky I don't know who is the owner for Microsoft.NET.Sdk.IL

**Comment by jkoritzinsky** [comment]
I think @tannergooding is the owner for the SDK, but I'm not positive.

**Comment by tannergooding** [comment]
I did the initial implementation work for `Microsoft.NET.Sdk.IL` but I'm not sure if I'm the "owner". Most of the changes since the initial work have been done by those working on infrastructure.

**Comment by ViktorHofer** [comment]
Just found this issue. That change was done by me when we consolidated the repositories. As I believed that dotnet/runtime is the only consumer of Microsoft.NET.Sdk.IL I didn't warn in beforehand or add a fallback mechanism... Sorry for that. To learn from this, next time I'll:
1. Gather potential consumers by scanning GitHub and devdiv/dnceng.
2. Open an issue to discuss the potential breaking change and loop in (ideally) all consumers.
3. Warn in beforehand / add a fallback mechanism (in this case it would have been as simple as accepting both properties).

Again sorry for the disruption here. For dotnet/runtime going forward we will use https://github.com/dotnet/runtime/issues/33821 in additional to what I said to announce breaking changes.

I think there's no further action necessary, hence closing.
